core/src/main/java/io/micronaut/core/annotation/ReflectionConfig.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.annotation;
+
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * An annotation that models directly the GraalVM reflect-config.json format.
+ *
+ * @since 3.5.0
+ * @author graemerocher
+ * @see io.micronaut.core.graal.GraalReflectionConfigurer
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Repeatable(ReflectionConfig.ReflectionConfigList.class)
+public @interface ReflectionConfig {
```

Comment:
Please, document it in [graalServices.adoc](https://github.com/micronaut-projects/micronaut-core/blob/3.4.x/src/main/docs/guide/languageSupport/graal/graalServices.adoc)